common/oauth.go

```diff
@@ -167,6 +181,15 @@ func (t *oauth2Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 		t.Debug(req2, cloneResponse(rsp))
 	}
 
+	// Certain providers return 401 when the token has been invalidated.
+	// This may indicate that the token needs to be forcefully refreshed.
+	// Since this is per-provider, the caller can provide a custom handler.
+	if rsp.StatusCode == http.StatusUnauthorized {
```

Comment:
I could be wrong, but I faintly remember SF sending an error back with an `invalid_grant` message but an empty response